pkg/collector/python/test_check.go

```diff
@@ -342,3 +345,10 @@ func testConfigureDeprecated(t *testing.T) {
 	assert.NotEqual(t, "", C.GoString(C.get_check_deprecated_agent_config))
 	assert.Equal(t, c.instance, C.get_check_deprecated_check)
 }
+
+func NewPythonFakeCheck() *PythonCheck {
+	c := NewPythonCheck("fake_check", nil)
+	// Remove check finalizer that may trigger race condition while testing
+	runtime.SetFinalizer(c, nil)
```

Comment:
Can we have the same issue outside uni tests?

Reply:
As far as I understand the rtloader logic we only create rtloader once for the entire lifetime of an agent process and if I'm right that particular problem cannot happen outside unit tests.

Reply:
So far yes that's the intended usage of the rtloader: only one instanciation and no dealloc+recreation, this issue should be scoped to only the UT.